src/kernel/src/fs/uio.rs

```diff
@@ -0,0 +1,96 @@
+use crate::errno::{Errno, EINVAL};
+use macros::Errno;
+use thiserror::Error;
+
+#[repr(C)]
+pub struct IoVec {
+    base: *const u8,
+    len: usize,
+}
+
+impl IoVec {
+    /// This is for when the PS4 DOES check the length (such as in recvmsg, recvfrom, sendmsg and sendto)
+    pub unsafe fn try_from_raw_parts(base: *const u8, len: usize) -> Result<Self, IoVecError> {
+        if len > IOSIZE_MAX {
+            return Err(IoVecError::MaxLenExceeded);
+        }
+
+        Ok(Self { base, len })
+    }
+
+    /// This is for when the PS4 DOES NOT check the length (such as in recvmsg, recvfrom, sendmsg and sendto)
```

Comment:
Those comments are confusing, e.g. does recvmsg check the length or not? (or both?)

Reply:
I copied it and forgot to change it, thanks